### PR TITLE
First commit for GSoC 2021: electron scattering

### PR DESCRIPTION
**Pull request by bsh90**

### CODE REVIEW

File: sympy/physics/mechanics/particle.py [hunk 1]
```diff
@@ -275,3 +278,11 @@ def parallel_axis(self, point, frame):
         from sympy.physics.mechanics import inertia_of_point_mass
         return inertia_of_point_mass(self.mass, self.point.pos_from(point),
                                      frame)
+
+    def electron_scattering(self, kinetic_energy, scattering_degree):
+        """By using Broglie relation and Lau condition, we can calculate the reciprocal lattic vector value"""
+        lambdaa = hbar/sqrt(2*m_e*kinetic_energy)
+        k = 2*pi/lambdaa
+        q = 2*k*sin(scattering_degree/2)
+        return q
```
[the remark this answers is not in the document]
Comment (bsh90) [reply sympy/physics/mechanics/particle.py:287]: This is the basic functions in solid state physics. this is not trivial! 
I wanted to add basic functions of some fields of physics. I can see there are lots of functions of quantum already.

**Comment by moorepants** [comment]
-1 to this as is. If you think it is appropriate for the mechanics module you'd need to supply a complete example showing why it is needed.

File: sympy/physics/quantum/constants.py [hunk 2]
```diff
@@ -57,3 +57,6 @@ def _latex(self, printer, *args):
 
 # Create an instance for everyone to use.
 hbar = HBar()
+
+#electron rest mass is 8.1871057769(25)×10−14 based on: https://en.wikipedia.org/wiki/Electron_rest_mass
+m_e = 8.1871057769*10**(−14)
```
[the remark this answers is not in the document]
Comment (bsh90) [reply sympy/physics/quantum/constants.py:62]: I added to that file with this pull request: https://github.com/sympy/sympy/pull/21215

**Comment by bsh90** [comment]
> -1 to this as is. If you think it is appropriate for the mechanics module you'd need to supply a complete example showing why it is needed.

Not mechanics, but it's electron scattering and it is used in solid state. There is no folder of solid state, nor statistical physics.

**Pull request closed without merging**